input/elasticapm/internal/modeldecoder/modeldecoderutil/http.go

```diff
@@ -45,18 +45,18 @@ func HTTPHeadersToStructPb(h http.Header) *structpb.Struct {
 	return nil
 }
 
-func HTTPHeadersToModelpb(h http.Header) []*modelpb.HTTPHeader {
+func HTTPHeadersToModelpb(h http.Header, out []*modelpb.HTTPHeader) []*modelpb.HTTPHeader {
```

Comment:
This doesn't seem right, we're passing out and also returning out?

Reply:
I find this API a bit confusing, I think it'd be best to do one or the other. If we want to accept the `out` as an argument, then we can pass `out *[]*modelpb.HTTPHeader`

Reply:
I guess this signature looks similar to `s = append(s, struct{})`, so that's an argument for this pattern. But we'll have to remember to use the return value.

Reply:
But for conversion purposes like this specific one, I think @marclop 's comment makes sense. It will be clearer.

Reply:
Thanks for chiming in. I was unaware of the widespread apdoption of this pattern in the current Go stdlib. I still think it's clearer and simpler to accept a single argument. If we leave it as is, I would prefer if we documented the behaviour since there are currently no godocs for these functions.